pkgs/tools/package-management/nix/default.nix

```
@@ -209,6 +209,13 @@ common =
     ];
   });
 
+  # master: https://github.com/NixOS/nix/pull/5536
+  # 2.4: https://github.com/NixOS/nix/pull/5537
+  installNlohmannJsonPatch = fetchpatch {
+    url = "https://github.com/NixOS/nix/pull/5536.diff";
```

```suggestion
    url = "https://github.com/NixOS/nix/commit/3884f7a69a57d8ecfcbcaae476ec2ff53ffbd549.patch";
```
that PR is not merged yet

---

True, but this solution was suggested by Eelco in an earlier (chaotic) iteration of the PR, https://github.com/NixOS/nix/pull/4904#issuecomment-858510744